dEdx topology correction: using threshold and qTot as input for qTot correction

matthias-kleiner opened the pull request:
- adding scaling of topology correction

matthias-kleiner committed:
dEdx topology correction: using threshold and qTot as input for qTot correction

wiechula commented on DataFormats/Detectors/TPC/include/DataFormatsTPC/CalibdEdxTrackTopologyPol.h:
```diff
@@ -67,16 +71,45 @@ class CalibdEdxTrackTopologyPol : public o2::gpu::FlatObject
+  /// \param charge charge of the cluster
+  GPUd() float getCorrection(const int region, const ChargeType chargeT, const float tanTheta, const float sinPhi, const float z, const float relPad, const float relTime, const float threshold, const float charge) const
+  {
+    const float corr = chargeT == ChargeType::Tot ? getCorrectionqTot(region, tanTheta, sinPhi, z, threshold, charge) : getCorrectionqMax(region, tanTheta, sinPhi, z, relPad, relTime);
```
Readability braces
```suggestion
    const float corr = (chargeT == ChargeType::Tot) ? getCorrectionqTot(region, tanTheta, sinPhi, z, threshold, charge) : getCorrectionqMax(region, tanTheta, sinPhi, z, relPad, relTime);
```

matthias-kleiner replied:
Thanks, fixed it along with some other minor changes

matthias-kleiner committed:
implementing scaling factor for CalibdEdxTrackTopologySpline class

- loading zero supression map only when using polynomial correction
- loading CalibdEdxTrackTopologySpline class from single object instead of from
multiple splines as input

matthias-kleiner committed:
dedx fixing pad position

matthias-kleiner commented:
The following file contains the new splines and polynomials for the correction [dedx_corrections.zip](https://github.com/AliceO2Group/AliceO2/files/8102329/dedx_corrections.zip)

[dE/dx vs tgl after correction with splines](https://github.com/AliceO2Group/AliceO2/files/8102332/tgl_spline.pdf)
[dE/dx vs tgl after correction with polynomials](https://github.com/AliceO2Group/AliceO2/files/8102334/tgl_pol.pdf)
